File: backend/src/main/java/com/zerofinance/xwallet/model/dto/LoginResponse.java
package com.zerofinance.xwallet.model.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * 登录响应DTO
 */
@Schema(description = "登录成功返回：JWT 与用户信息")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LoginResponse {
    @Schema(description = "JWT，请求需认证的接口时在 Header 填 Authorization: Bearer <token>")
    private String token;
    @Schema(description = "用户基本信息")
    private UserInfo userInfo;

    @Schema(description = "用户基本信息")
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class UserInfo {
        private Long userId;
        private String username;
        @Schema(description = "SYSTEM / CUSTOMER")
        private String userType;
        @Schema(description = "系统用户的角色编码列表，顾客为空")
        private List<String> roles;
    }
}

File: app/lib/models/login_response.dart
/// 登录响应数据模型
class LoginResponse {
  final String token;
  final UserInfo userInfo;

  LoginResponse({
    required this.token,
    required this.userInfo,
  });

  /// 从JSON创建实例
  factory LoginResponse.fromJson(Map<String, dynamic> json) {
    return LoginResponse(
      token: json['token'] ?? '',
      userInfo: UserInfo.fromJson(json['userInfo'] ?? {}),
    );
  }
}

/// 用户信息
class UserInfo {
  final int userId;
  final String username;
  final String userType; // "SYSTEM" 或 "CUSTOMER"
  final String? role; // 仅系统用户有值

  UserInfo({
    required this.userId,
    required this.username,
    required this.userType,
    this.role,
  });

  /// 从JSON创建实例
  factory UserInfo.fromJson(Map<String, dynamic> json) {
    return UserInfo(
      userId: json['userId'] ?? 0,
      username: json['username'] ?? '',
      userType: json['userType'] ?? '',
      role: json['role'],
    );
  }
}

/// 通用响应结果
class ResponseResult<T> {
  final int? code;
  final String? message;
  final T? data;

  ResponseResult({
    this.code,
    this.message,
    this.data,
  });

  /// 从JSON创建实例
  factory ResponseResult.fromJson(Map<String, dynamic> json, T Function(dynamic)? dataParser) {
    return ResponseResult(
      code: json['code'],
      message: json['message'],
      data: dataParser != null && json['data'] != null ? dataParser(json['data']) : json['data'],
    );
  }

  /// 是否成功
  bool get isSuccess => code == null || code == 200;
}
